snuba/admin/clickhouse/predefined_system_queries.py

```diff
@@ -0,0 +1,120 @@
+from dataclasses import dataclass
+from typing import Dict, Optional, Sequence, Type
+
+
+class _QueryRegistry:
+    """Keep a mapping of SystemQueries to their names"""
+
+    def __init__(self) -> None:
+        self.__mapping: Dict[str, Type["SystemQuery"]] = {}
+
+    def register_class(self, cls: Type["SystemQuery"]) -> None:
+        existing_class = self.__mapping.get(cls.__name__)
+        if not existing_class:
+            self.__mapping[cls.__name__] = cls
+
+    def get_class_by_name(self, cls_name: str) -> Optional[Type["SystemQuery"]]:
+        return self.__mapping.get(cls_name)
+
+    @property
+    def all_queries(self) -> Sequence[Type["SystemQuery"]]:
+        return list(self.__mapping.values())
+
+
+_QUERY_REGISTRY = _QueryRegistry()
+
+
+@dataclass
+class SystemQuery:
+    sql: str
+
+    @classmethod
+    def to_json(cls) -> Dict[str, Optional[str]]:
+        return {
+            "sql": cls.sql,
+            "description": cls.__doc__,
+            "name": cls.__name__,
+        }
+
+    def __init_subclass__(cls) -> None:
+        _QUERY_REGISTRY.register_class(cls)
+        return super().__init_subclass__()
+
+    @classmethod
+    def from_name(cls, name: str) -> Optional[Type["SystemQuery"]]:
+        return _QUERY_REGISTRY.get_class_by_name(name)
+
+    @classmethod
+    def all_queries(cls) -> Sequence[Type["SystemQuery"]]:
+        return _QUERY_REGISTRY.all_queries
+
+
+class CurrentMerges(SystemQuery):
+    """Currently executing merges"""
+
+    sql = """
+    SELECT
+        count(),
```

Comment:
`SELECT table, elapsed, progress, result_part_name`
This will be more useful. The number of merges without knowing what clickhosue is merging is not useful.